### PR TITLE
feat(release): introduce "release" task

### PR DESCRIPTION
**Pull request by eladb**
Extract some steps from the GitHub release workflow into a projen task so it can be executed manually via `projen release`.

The `release` task **prepares a release** by performing the following: `bump` + `build` + `unbump` + anti-tamper. 

The task will emit two files under the `dist/` directory which can be used to publish a release: `version.txt` and `changelog.md` (both are .gitignored). The GitHub workflow will pick up those files (and the rest of the contents under `dist`) and publish them as GitHub releases and to the various package managers.

Next steps: this is an initial step to address #782 and #726. The next step is to add tasks that can be used to publish a release. These tasks can be composed of `publish:github`, `publish:npm`, `publish:nuget`, which we can then use in the GitHub workflow. Additionally, we will likely want to allow manual creation of git tags and committing of a CHANGELOG file when projects are not hosted on GitHub.

Misc: add more information to task inspection and execution outputs (namely, include environment variables).

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,6 +64,7 @@
     "standard-version": "^9",
     "typescript": "^3.9.10"
   },
+  "peerDependencies": {},
```
Comment (eladb) [inline package.json:67]:
```suggestion
```

**Comment by eladb** [comment]
@Chriscbr can you take a quick look to see I didn't mess this up?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -13,7 +13,6 @@ jobs:
       contents: write
     env:
       CI: "true"
-      RELEASE: "true"
```
Comment (Chriscbr) [inline .github/workflows/release.yml:16]: "release" is the only task that uses this yes?

File: src/tasks/runtime.ts [hunk 3]
```diff
@@ -158,18 +170,25 @@ class RunTask {
...
+  private resolveEnvironment(parents: string[]) {
+    let env = this.runtime.manifest.env ?? {};
+
+    // add env from all parent tasks one by one
+    for (const parent of parents) {
+      env = {
+        ...env,
+        ...this.runtime.tryFindTask(parent)?.env ?? {},
+      };
+    }
```
Comment (Chriscbr) [inline src/tasks/runtime.ts:183]: A unit test for this would be nice to have

**Review by Chriscbr — APPROVED** [review]
+1 on moving towards modeling github workflows as much as possible within projen tasks.

Terminology question - what's the difference between "releasing" and "publishing"? It feels to me as though the codebase uses them interchangeably, but it's not really clear since we have have public APIs using these names (`Release` and `Publisher`).

Related, it looks like we are dividing releasing/publishing into a two step process (preparing the files for distribution, and then posting them on npm/maven/etc.). If the "release" task doesn't actually release, maybe another name would be better (e.g. "prepare-release", "publish:prepare")?

~Can you add some info in `docs/` about projen's manual release features/options?~ 😄 edit: This part hasn't been finished yet, so nvm

Looks great!

File: .github/workflows/release.yml (hunk 2 above)
Comment (eladb) [reply .github/workflows/release.yml:16]: Yap.

File: src/tasks/runtime.ts (hunk 3 above)
Comment (eladb) [reply src/tasks/runtime.ts:183]: Done.

**Pull request merged**